src/components/views/admin/BuilderView.tsx

```diff
@@ -1,156 +1,243 @@
+        );
+      } else {
+        candidatesCv[memberId] = {
+          [type]: typeDataByMemberId[memberId].map((item) => item.data),
+        };
+      }
+    });
+  });
+
+  const [ratings, setRatings] = useState<{ [key: string]: number | null }>({});
+  const handleRatingChange = (
+    memberId: string,
+    newRating: number | null,
+  ): void => {
+    const rateData = appDataArray.find(
+      (data) =>
+        data.data.memberId === memberId && data.type === 'candidateRateInfo',
+    );
+    if (rateData) {
+      handlePatch(rateData.id, {
+        ...rateData.data,
+        rating: newRating,
+        memberId,
+      });
+    } else {
+      handlePost({ memberId, rating: newRating });
+    }
+    setRatings((prevRatings) => ({
+      ...prevRatings,
+      [memberId]: newRating,
+    }));
+  };
+
+  const handleDownload = async (memberId: string): Promise<void> => {
+    const candidateCvData = candidatesCv[memberId];
+    const { component: CvTemplate } = TEMPLATES.find(
+      (t) => t.id === candidateCvData.cvStatusInfo.selectedTemplateId,
+    ) || { component: FirstTemplate };
+    const renderedTemplate = <CvTemplate cvValues={candidateCvData} />;
+    const pdfBlob = await pdf(renderedTemplate).toBlob();
+    if (!candidateCvData.cvStatusInfo.customCv) {
+      saveAs(pdfBlob, 'generated-cv.pdf');
+    }
+  };
+  const [visibility, setVisibility] = useState<string | null>(null);
+  const [renderedFile, setRenderedFile] = useState<string | null>(null);
+  const handleView = async (memberId: string): Promise<void> => {
+    if (visibility === memberId) {
+      setVisibility(null);
+      setRenderedFile(null);
+    } else {
+      const candidateCvData = candidatesCv[memberId];
+      const { component: CvTemplate } = TEMPLATES.find(
+        (t) => t.id === candidateCvData.cvStatusInfo.selectedTemplateId,
+      ) || { component: FirstTemplate };
+      const fileToRender = <CvTemplate cvValues={candidateCvData} />;
+      const pdfBlob = await pdf(fileToRender).toBlob();
+      const dataUrl = URL.createObjectURL(pdfBlob);
+      setVisibility(memberId);
+      setRenderedFile(dataUrl);
+      window.open(dataUrl, '_blank');
+    }
+  };
+  const handleDownloadAll = async (): Promise<void> => {
+    const zip = new JSZip();
+    const candidateIds = Object.keys(candidatesCv);
+
+    // Use Promise.all with map to wait for all async operations to complete
+    await Promise.all(
+      candidateIds.map(async (memberId) => {
+        const candidateCvData = candidatesCv[memberId];
+        const { component: CvTemplate } = TEMPLATES.find(
+          (t) => t.id === candidateCvData.cvStatusInfo.selectedTemplateId,
+        ) || { component: FirstTemplate };
+        const renderedTemplate = <CvTemplate cvValues={candidateCvData} />;
+        const pdfBlob = await pdf(renderedTemplate).toBlob();
+        if (!candidateCvData.cvStatusInfo.customCv) {
+          zip.file(`${memberId}-generated-cv.pdf`, pdfBlob);
+        }
```

Comment:
And what if they have a custom cv uploaded ? you don't include it in the complete zip ?

Reply:
As mentioned in a previous comment, it needs to be handled for both cases. I will implement that.